megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java

```diff
@@ -0,0 +1,470 @@
+                    // our current shortest off-board path and keep it in mind if it's both shorter and safe
+                    // at the end of the process, we will add the shortest path off board to the list of moves
+                    if(shortestPath == null || 
+                            (shortestPath.length() > e.length()) && 
+                            AeroPathUtil.isSafePathOffBoard(e)) {
+                        shortestPath = e;
+                    }
+                }
+            }
+            
+            return filteredMoves;
+        }
+        
+        /**
+         * Tests if the object should stay in the collection.
+         * 
+         * @param object tested object
+         * @return true if the object should stay in the collection
+         */
+        @Override
+        public boolean shouldStay(MovePath path) {
+            return !path.fliesOffBoard();
+        }
+        
+    }
+
+    // This object contains a hex and the shortest non-off-board path that has flown over it.
+    HashMap<Coords, MovePath> visitedHexes;
+    
+    /**
+     * Generates paths that begin with all valid acceleration sequences for this aircraft.
+     * Avoids sequences that will cause a PSR (just no)
+     * @param startingPath
+     * @return
+     */
+    public Collection<MovePath> generateValidAccelerations(MovePath startingPath)
+    {
+        // the possible velocities at ground level are 1 - 12 (we ignore "0" velocity as that will just stall our aircraft)
+        // with an additional lower and upper bound of "current velocity" -/+ walk MP (investigate utility of adjusting it to be run MP instead)
+
+        Collection<MovePath> paths = new ArrayList<MovePath>();
+        
+        // sanity check: if we've already done something else with the path, there's no acceleration to be done
+        if(startingPath.length() > 0) {
+            return paths;
+        }
+        
+        int currentVelocity = startingPath.getFinalVelocity();
+        int lowerBound = Math.max(1, currentVelocity - maxThrust);
+        // put a governor on that engine
+        // while the rulebook max velocity on ground map is 12, in reality we want to limit ourselves to velocity 3, for two reasons:
+        // 1: velocity 3 lets us do a 360 degree turn, so any further turning is superfluous
+        // 2: velocity 4 generates about 6,000 paths which is pretty painful for performance, although we can probably improve
+        //      that by updating PathRanker.getMovePathSuccessProbability() to skip most non-aero related checks
+        int upperBound = Math.min(3, currentVelocity + maxThrust); 
+        
+        // we go from the lower bound to the current velocity and generate paths with the required number of DECs to get to
+        // the desired velocity
+        for(int desiredVelocity = lowerBound; desiredVelocity < currentVelocity; desiredVelocity++)
+        {
```

Comment:
`for(int desiredVelocity = lowerBound; desiredVelocity < currentVelocity; desiredVelocity++) {`